### DISCUSSION TITLE
Question about InputProperties and AffectedProperties in Business Rule

### DISCUSSION THREAD

**Discussion by Chicagoan2016** [MarimerLLC/csla#3361] [Questions] [answered]
I am going through old code and I noticed something about the way we have called the Add method of InputProperties and AffectedProperties inside our rules constructor.

In some cases, we have supplied the parameter name to InputProperties.Add like
  ``` cs
     InputProperties.Add(employeeid)  //employeeid is a parameter of our business rule
  ```
and in other cases
we have something like

 ```cs
    InputProperties.Add(EmployeeID) //EmployeeID is the property defined in our Business rule class
  ```

Is there any difference between the two?

Kind regards

**Comment by rockfordlhotka** [comment, marked as answer]
If you are creating a rule that is designed for reuse across any type of business class, then you need to pass properties as parameters to the constructor.

If you are creating a private rule for one specific business class, then you can hard-code the properties into the rule. Personally I try to nest these rules inside the business class to make it clear that they are "private". You can also mark them `private`, but that makes it harder to write unit tests for the rule.

**Reply by Chicagoan2016** [reply]
Thank you @rockfordlhotka 
so we should use 
    ```cs
        InputProperties.Add(PrimaryProperty);
     ```
instead of
    ```cs
          InputProperties.Add(primaryproperty);
     ```
?

**Reply by rockfordlhotka** [reply]
That's really a style question - either will work in the constructor.
